djangosaml2idp/views.py

```diff
@@ -128,6 +165,10 @@ def get(self, request, *args, **kwargs):
             sp_config = settings.SAML_IDP_SPCONFIG[resp_args['sp_entity_id']]
         except Exception:
             return self.handle_error(request, exception=ImproperlyConfigured("No config for SP %s defined in SAML_IDP_SPCONFIG" % resp_args['sp_entity_id']), status=400)
+        user_identity = getattr(
+            settings, 'SAML_IDP_IDENTIFYING_ATTRIBUTE', 'username')
+        if sp_config.get('NameIDAttr', None):
+            user_identity = sp_config.get('NameIDAttr')
```

Comment:
The 4 lines above (assuming you reuse the `user_identity` on top) can be a single-line:

```suggestion
            user_identity = sp_config.get('NameIDAttr', user_identity)
```